### PR TITLE
#155394319 - Set errors in the store

### PR DESCRIPTION
**Pull request by peterpaints**
#### What does this PR do?
Adds an Incident Error action creator that is dispatched when the loadIncidents thunk returns
an error.
#### Description of Task to be completed?
- We need to get the error in the state and show it to the user.
- When an error occurs, an appropriate action should be dispatched to set the error in the store to be displayed to the user.
#### Where should the reviewer start?
View file changes on PR.
#### How should this be manually tested?
Manually modify the mock api to return an error. For instance, change one of the functions in mockMiddleware.js from say:
```
fetchIncidents: (req, res) => {
    setTimeout(() => {
      res.send({ data: { incidents: getIncidents() }, status: 'success' });
    }, 2000);
  }
```
to:
```
fetchIncidents: (req, res) => {
    setTimeout(() => {
      res.status(404).send({ error: 'Not found', status: 'error' });
    }, 2000);
  }
```
Attempt to reload the dashboard. A custom toast message should appear with the error status code and error message. Also use redux dev tools to see if the incident error action creator was 
dispatched.
#### What are the relevant pivotal tracker stories?
#155394319
#### Screenshots:
screen shot 2018-03-01 at 11 36 59 am
screen shot 2018-03-04 at 3 09 43 pm

### CODE REVIEW

**Comment by kimobrian** [comment]
@peterpaints I thought I left a comment here but can't see it. Would you please add some more color on how to modify the API so as to get errors. It will be easier to test that way. Also sync with Akash to combine your work.

**Comment by peterpaints** [comment]
@kimobrian I have been mocking the API in my tests using the moxios package. What this package does is it intercepts any API calls made in my tests, and returns whatever I configure it to return. So in the case of an error I have it returning a sample error response, then checking that the INCIDENT_ERROR action creator was dispatched in the mock store.

**Comment by kimobrian** [comment]
@peterpaints  I meant manual testing like, you could say: `Change line x to y and reload the page to see the error`. Writing tests might ensure that the product meets specifications but does not always guarantee that it meets the user requirements. Something else, do the errors get displayed or are they just in the store because it will make more sense if they are displayed(that's more like what the user will want).

**Comment by peterpaints** [comment]
@kimobrian I understand. Well, for manual testing, one could change one of the functions in mockMiddleware.js from say
```
fetchIncidents: (req, res) => {
    setTimeout(() => {
      res.send({ data: { incidents: getIncidents() }, status: 'success' });
    }, 2000);
  }
```
to:
```
fetchIncidents: (req, res) => {
    setTimeout(() => {
      res.status(404).send({ error: 'Not found', status: 'error' });
    }, 2000);
  }
```
For now, errors are displayed in the store. Let me change that so that they are displayed to the user. I'm doing that ASAP.
I'll also add the example for simulating errors above to the PR under `How can this be manually tested`.

**Comment by peterpaints** [comment]
@RayNjeri How about now? I have rebased.

**Comment by NaiRobley** [comment]
@kimobrian we have a custom notification component that displays errors or success messages to the user as snack-bars. It would be ideal for the errors

**Comment by kimobrian** [comment]
@peterpaints See @NaiRobley 's comment above 👆

File: package.json [hunk 1]
```diff
@@ -43,7 +44,11 @@
         "react-router-dom": "^4.2.2",
         "react-router-redux": "^4.0.8",
         "redux": "^3.7.2",
-        "redux-thunk": "^2.2.0"
+        "redux-mock-store": "^1.5.1",
+        "redux-thunk": "^2.2.0",
```
Comment (kimobrian) [inline package.json]: Refer to this [PR](https://github.com/AndelaOSP/wire/pull/45) on package.json clean up.`moxios` is also a devDependency.

**Comment by peterpaints** [comment]
@kimobrian I synced with him and implemented it. I have added changes to the PR description to reflect that. Including appropriate screenshots.

File: package.json (hunk 1 above)
Comment (peterpaints) [reply package.json]: Now fixed.

**Comment by kimobrian** [comment]
@peterpaints Is that the error message `404 Not Found`? Let's find a way to have better error messages that are user friendly like `The incident record was not found` or if not sure what happened you can say `An error occurred while fetching the incident`

**Comment by peterpaints** [comment]
@kimobrian Certainly. I have fixed the error messages, and replaced the screenshot above.

**Pull request merged**